### PR TITLE
Only rebuild shaders on sky rotation change if changing away from identity

### PR DESCRIPTION
**Pull request by mvaligursky**
Fixes https://github.com/playcanvas/engine/issues/5458

Don't rebuild shaders if rotation changes, only if it changes and it was identity before, which requires shaders to be re-generated to include support for rotation, which is excluded when identity.

### CODE REVIEW

File: src/scene/scene.js [hunk 1]
```diff
@@ -626,14 +626,23 @@ class Scene extends EventHandler {
...
                 this._skyboxRotationMat3.setIdentity();
             } else {
                 this._skyboxRotationMat4.setTRS(Vec3.ZERO, value, Vec3.ONE);
                 this._skyboxRotationMat4.invertTo3x3(this._skyboxRotationMat3);
             }
-            this._resetSky();
+
+            // only reset sky / rebuild scene shaders if rotation changed from identity, to include
+            // support for skybox rotation in shaders
+            if (prevIdentity && !newIdentity) {
```
Comment (slimbuck) [inline src/scene/scene.js:643]: This should either do `prevIdentity !== newIdentity` or scene should store a flag `skyboxHasRotation` which tracks first time rotation is set (and then doesn't reset again).

Comment (mvaligursky) [reply src/scene/scene.js:643]: I don't want to rebuild shaders if we go from rotation to no rotation.

Comment (slimbuck) [reply src/scene/scene.js:643]: Right, but now you're rebuilding every time you go no->rotation. I think you want a flag that says 'we've compiled rotation' and only do that once.

Comment (mvaligursky) [reply src/scene/scene.js]: ok this is done, thanks

**Pull request merged**